xbmc/utils/GroupUtils.cpp

```diff
@@ -20,13 +20,13 @@
 
 using SetMap = std::map<int, std::set<CFileItemPtr> >;
 
-bool GroupUtils::Group(GroupBy groupBy, const std::string &baseDir, const CFileItemList &items, CFileItemList &groupedItems, GroupAttribute groupAttributes /* = GroupAttributeNone */)
+bool GroupUtils::Group(GroupBy groupBy, const std::string &baseDir, const CFileItemList &items, CFileItemList &groupedItems, int groupRating, GroupAttribute groupAttributes /* = GroupAttributeNone */)
```

Comment:
I would prefer it if we would use a
```cpp
enum class GroupRatingBy
{
  Average,
  Highest
};
```
and use that instead of an ambiguous `int`. And it could later be re-used for other groupings. You will obviously have to map the value of the newly introduced setting to the `GroupRatingBy` before calling `GroupUtils::Group()`.
And I would move it to the end of the argument list and default it to `GroupRatingBy::Average`.